### PR TITLE
[Fix 4488] Ensure that corrupted DB entries are processed gracefully

### PR DESCRIPTION
**Pull request by muffins**
This addresses #4488. On windows we've been seeing high volumes of crashes generating from queries against the Windows Event Log publisher. What we found was that there existed corrupted values of eventid:event_time record entries, which were being de-referenced incorrectly, causing access violation crashes. These crashes continued to occur, which put the daemon into a start -> record events -> crash loop, this the events were never flushed from the DB, and we ended up seeing inflated database sizes, on the GB scale.

This ensures that when we encounter corrupted DB record entries we skip over them and warn, as opposed to crashing the daemon.

### CODE REVIEW

File: osquery/events/events.cpp [hunk 1]
```diff
@@ -401,16 +401,20 @@ std::vector<EventRecord> EventSubscriberPlugin::getRecords(
...
-      boost::split(bin_records, record_value, boost::is_any_of(",:"));
+      bin_records = split(record_value, ",");
     }
 
-    auto bin_it = bin_records.begin();
     // Iterate over every 2 items: EID:TIME.
-    for (; bin_it != bin_records.end(); bin_it++) {
-      const auto& eid = *bin_it;
-      EventTime et = timeFromRecord(*(++bin_it));
+    for (size_t i = 0; i < bin_records.size(); i++) {
```
Comment (theopolis) [inline osquery/events/events.cpp]: If you do not need the index, then `for (const auto& record : bin_records)`.

File: osquery/events/events.cpp [hunk 2]
```diff
@@ -401,16 +401,20 @@ std::vector<EventRecord> EventSubscriberPlugin::getRecords(
...
+      auto vals = split(bin_records[i], ":");
+      if (vals.size() != 2) {
+        LOG(WARNING) << "Event records mismatch, " << bin_records[i]
```
Comment (theopolis) [inline osquery/events/events.cpp]: nit, I prefer `:` over commas, but your choice.

File: osquery/events/events.cpp [hunk 3]
```diff
@@ -401,16 +401,20 @@ std::vector<EventRecord> EventSubscriberPlugin::getRecords(
...
+      bin_records = split(record_value, ",");
     }
 
-    auto bin_it = bin_records.begin();
     // Iterate over every 2 items: EID:TIME.
-    for (; bin_it != bin_records.end(); bin_it++) {
-      const auto& eid = *bin_it;
-      EventTime et = timeFromRecord(*(++bin_it));
+    for (const auto& record : bin_records) {
+      auto vals = split(record, ":");
```
Comment (akindyakov) [inline osquery/events/events.cpp]: You have no intentions to change `vals`, it a good reason to mark it `const` :)

File: osquery/events/events.cpp [hunk 4]
```diff
@@ -401,16 +401,20 @@ std::vector<EventRecord> EventSubscriberPlugin::getRecords(
...
+      if (vals.size() != 2) {
+        LOG(WARNING) << "Event records mismatch: " << record
+                     << " does not have a matching eid/event_time";
+        continue;
+      }
+      const auto eid = vals[0];
+      EventTime et = timeFromRecord(vals[1]);
```
Comment (akindyakov) [inline osquery/events/events.cpp]: You have no intentions to change `et`, it a good reason to mark it `const` :)

**Pull request merged**